confignode/src/main/java/org/apache/iotdb/confignode/service/thrift/ConfigNodeRPCServiceProcessor.java

```diff
@@ -466,6 +476,57 @@ public long getConfigNodeHeartBeat(long timestamp) throws TException {
     return timestamp;
   }
 
+  @Override
+  public TShowDataNodesResp showDataNodes(TShowDataNodesReq showDataNodesReq) throws TException {
+    GetRegionInfoListPlan getRegionsInfoPlan =
+        new GetRegionInfoListPlan(showDataNodesReq.getConsensusGroupType());
+    Map<Integer, List<Integer>> map = new HashMap<>();
```

Comment:
Could we use a counter to count dynamically? instead of adding all into a list and then count..

Reply:
ok ,I reconstructed the counting method